integration_tests/src/bin/benchmark/main.rs

```diff
@@ -0,0 +1,302 @@
+            });
+
+            let mut arguments = arguments.clone();
+            arguments[i] = Argument::Constant(result);
+            reports.push(args.with_arguments(&arguments));
+            return true;
+        }
+    }
+
+    // if we got here, there is no maximize, so just run a single test
+    let config = args.with_arguments(&arguments);
+    run_single_test(ctx, config)
+}
+
+fn binsearch(mut start: u64, mut end: u64, test_fn: impl Fn(u64) -> bool) -> u64 {
+    while start < end {
+        let midpoint = (start + end + 1) / 2;
+
+        if test_fn(midpoint) {
+            start = midpoint;
+        } else {
+            end = midpoint - 1;
+        }
+    }
+
+    end
+}
+
+fn find_upper_bound(start: u64, test_fn: impl Fn(u64) -> bool) -> u64 {
+    let mut end = start;
+
+    while test_fn(end) {
+        end *= 2;
+    }
+
+    end - 1
+}
+
+/// true - works
+/// false - too slow
+fn run_single_test(ctx: GraphicsContext, bench_config: SingleBenchConfig) -> bool {
+    let (pipeline, _event_loop) = Pipeline::new(Options {
+        queue_options: QueueOptions {
+            never_drop_output_frames: true,
+            output_framerate: Framerate {
+                num: bench_config.framerate as u32,
+                den: 1,
+            },
+            default_buffer_duration: queue::DEFAULT_BUFFER_DURATION,
+            ahead_of_time_processing: false,
+            run_late_scheduled_events: true,
+        },
+        web_renderer: WebRendererInitOptions {
+            enable: false,
+            enable_gpu: false,
+        },
+        wgpu_ctx: Some(ctx),
+        force_gpu: false,
+        download_root: std::env::temp_dir(),
+        wgpu_features: wgpu::Features::UNIFORM_BUFFER_AND_STORAGE_TEXTURE_ARRAY_NON_UNIFORM_INDEXING
+            | wgpu::Features::SAMPLED_TEXTURE_AND_STORAGE_BUFFER_ARRAY_NON_UNIFORM_INDEXING,
+        load_system_fonts: Some(false),
+        output_sample_rate: 48_000,
+        stream_fallback_timeout: Duration::from_millis(500),
+        tokio_rt: None,
+        stun_servers: Vec::new().into(),
+    })
+    .unwrap();
+    let pipeline = Arc::new(Mutex::new(pipeline));
+
+    let mut inputs = Vec::new();
+    for i in 0..bench_config.decoder_count {
+        let input_id = InputId(format!("input_{i}").into());
+        inputs.push(input_id.clone());
+
+        Pipeline::register_input(
+            &pipeline,
+            input_id,
+            RegisterInputOptions {
+                input_options: InputOptions::Mp4(Mp4Options {
+                    should_loop: true,
+                    video_decoder: bench_config.video_decoder,
+                    source: Source::File(bench_config.file_path.clone()),
+                }),
+                queue_options: QueueInputOptions {
+                    offset: Some(Duration::ZERO),
+                    required: true,
+                    buffer_duration: None,
+                },
+            },
+        )
+        .unwrap();
```

Comment:
I came across an error 
```
InputError(InputId("input_121"), Mp4(IoError(Os { code: 24, kind: Uncategorized, message: "Too many open files" })))
```
It should be handled somehow imo. Maybe just log it and assume that last count was max? I think it would be pretty anoying when I got such error after running test for longer time.

Maybe make `run_single_test` return `Result<bool, some-kind-of-error>` and handle those on higher level?